src/app/utils/date-utils.ts

```diff
@@ -15,5 +15,14 @@ export function getHttpFriendly(date: moment.Moment): string {
 }
 
 export function formatDuration(duration: moment.Duration): string {
-    return moment.utc(moment.duration(duration).asMilliseconds()).format('HH:mm');
+    const durationMs = duration.asMilliseconds();
+    if (durationMs >= 86400000) {
```

Comment:
This number should probably be extracted as a const e.g. durationOfDayInMs or some better name.